### DISCUSSION TITLE
How to create a Custom Form Field In filament using relational column.

### DISCUSSION THREAD

**Discussion by DimuthuPHD** [filamentphp/filament#6123] [General]
I'm currently in the process of developing a content management system (CMS) that will handle education migrations for various countries. This CMS includes tables for programs and languages. When creating a new program, I need to incorporate program duration for each language. I'm considering either creating a custom field or utilizing an existing field to achieve this functionality. Can you provide guidance to how to do this ?

**Comment by danharrin** [comment]
You can embed custom fields within https://filamentphp.com/docs/2.x/forms/layout#saving-data-to-relationships to access relational data

**Comment by tauseedzaman** [comment]
![image](https://github.com/filamentphp/filament/assets/64689921/05951d68-e1ea-4b83-9a9f-36569bbef08b)
i want a one input field in each row how to achive this

**Reply by danharrin** [reply]
wrap each input in a Grid with `columnSpanFull()`
